packages/atclient/src/connection.c

```diff
@@ -344,6 +432,10 @@ void atclient_connection_free(atclient_connection *ctx) {
   memset(ctx->host, 0, ATCLIENT_CONSTANTS_HOST_BUFFER_SIZE);
   ctx->port = -1;
   ctx->should_be_connected = false;
+
+  if (ctx->hooks != NULL) {
+    free(ctx->hooks);
+  }
```

Comment:
FYI here is where the hooks are freed, this is allocated by `atclient_connection_enable_hooks`

Reply:
you don't need to worry about the memory ctx->hooks->X since they are function pointers.